src/station_cmd.cpp

```diff
@@ -1834,18 +1835,30 @@ static CommandCost FindJoiningRoadStop(StationID existing_stop, StationID statio
  */
 CommandCost CmdBuildRoadStop(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32 p2, const char *text)
 {
-	bool type = HasBit(p2, 0);
-	bool is_drive_through = HasBit(p2, 1);
-	RoadType rt = Extract<RoadType, 5, 6>(p2);
+	bool is_drive_through = HasBit(p2, 23);
+	RoadType rt = Extract<RoadType, 16, 6>(p2);
+	//RoadType rt = Extract<RoadType, 5, 6>(p2);
```

Comment:
I guess regression fails because the changes here are not applied to script API.

Reply:
In which way have these to be documented or implemented?

Reply:
The code in the script API implementation that packs the parameters is separate from the code in the GUI that packs the parameters. make sure the script API packs parameters the correctly. Probably this:
https://github.com/spnda/OpenTTD/blob/e27250c4ee38708c860d513ba09266b43dc8c173/src/script/api/script_road.cpp#L534-L559